vuln-reach/src/javascript/lang/symbol_table.rs

```diff
@@ -299,24 +308,39 @@ impl<'a> SymbolTable<'a> {
 
         let name = self.tree.repr_of(node);
 
-        // If the identifier is in a formal parameters list, the declaration
-        // scope is in the sibling "body" field block.
+        // For function parameters, the scope is the function body.
         let mut parent = cursor.goto_parent().unwrap();
         if parent.kind() == "formal_parameters" {
-            // If formal_parameters nodes has a statement_block next sibling, it is a
-            // regular function, otherwise it is an arrow function and the
-            // identifier doesn't belong to a scope.
-
+            // Get function body.
             let grandparent = cursor.clone().goto_parent().unwrap();
             let body = grandparent.child_by_field_name("body").unwrap();
+
             if body.kind() == "statement_block" {
-                let scope_index = *self.scope_indices.get(&body).unwrap();
-                let scope = &self.scopes[scope_index];
+                // Get scope and ensure the identifier is defined in it.
+                let scope = self.get_scope(body).unwrap();
                 assert!(scope.names.iter().any(|&node| self.tree.repr_of(node) == name));
+
                 return Some((scope, node));
+            } else {
+                // Lambda parameters without body do not belong to any scope.
+                //
+                // Example: `(param) => console.log(param);`
+                return None;
             }
         }
 
+        // For parameters in a catch clause, the scope is the catch body.
+        if parent.kind() == "catch_clause" {
+            // Get catch body.
+            let body = parent.child_by_field_name("body").unwrap();
+
+            // Get scope and ensure the identifier is defined in it.
+            let scope = self.get_scope(body).unwrap();
+            assert!(scope.names.iter().any(|&node| self.tree.repr_of(node) == name));
```

Comment:
I'd like to leave them in just as a way of holding invariants/"documenting" expected behavior, but I agree that that could be cumbersome (not to mention stop-the-world makes using the library more unpredictable from the client's perspective).

I'd say we could move to using `debug_assert` everywhere instead of plain `assert`, so we only pay the price during testing.

Reply:
Let's do this separately then.